62.CAD/Hatch.h

```diff
@@ -0,0 +1,75 @@
+
+#include "../common/CommonAPI.h"
+#include "nbl/ext/FullScreenTriangle/FullScreenTriangle.h"
+#include "nbl/core/SRange.h"
+#include "glm/glm/glm.hpp"
+#include <nbl/builtin/hlsl/cpp_compat/matrix.hlsl>
+#include <nbl/builtin/hlsl/cpp_compat/vector.hlsl>
+#include "curves.h"
+
+using namespace nbl;
+
+namespace hatchutils {
+	static constexpr double QUARTIC_THRESHHOLD = 1e-10;
+
+	static std::array<double, 4> solveQuarticRoots(double a, double b, double c, double d, double e, double t_start, double t_end);
+    
+	static float64_t2 getCurveRoot(double p0, double p1, double p2);
+}
+
+class Hatch
+{
+public:
+	// this struct will be filled in cpu and sent to gpu for processing as a single DrawObj
+	struct CurveHatchBox
+	{
+		float64_t2 aabbMin, aabbMax;
+		float64_t2 curveMin[3];
+		float64_t2 curveMax[3];
+	};
+
+	// TODO: start using A, B, C here
```

Comment:
Remove TODO. not worth it here cause our intersection math heavily relies on this form and changing it requires 2-days full worth of math from me